sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -287,7 +287,8 @@ trait RuntimeReplaceable extends UnaryExpression with Unevaluable {
 trait NonSQLExpression extends Expression {
   final override def sql: String = {
     transform {
-      case a: Attribute => new PrettyAttribute(a)
+      case a: Attribute => new PrettyNamedExpression(a)
+      case a: Alias => new PrettyNamedExpression(a.sql, a.dataType)
```

Comment:
`case a: NamedExpression => new PrettyNamedExpression(a)`

Move all the processing logics in `PrettyNamedExpression `